inc/c_util/two_d_array_ll.h

```diff
@@ -15,10 +15,236 @@
+        LogError("Invalid arguments: TWO_D_ARRAY_HANDLE two_d_array_handle=%p", two_d_array);                                                                       \
+        result = MU_FAILURE;                                                                                                                                        \
+    }                                                                                                                                                               \
+    else                                                                                                                                                            \
+    {                                                                                                                                                               \
+        /* Codes_SRS_TWO_D_ARRAY_07_011: [ If row_index is equal or greater than row_size, TWO_D_ARRAY_FREE_ROW(T) shall fail and return a non-zero value. ]*/      \
+        if (row_index >= two_d_array->rows)                                                                                                                         \
+        {                                                                                                                                                           \
+            LogError("Invalid arguments: uint64_t row_index=%" PRIu64 " out of bound, total_rows = %" PRIu64, row_index, two_d_array->rows);                       \
+            result = MU_FAILURE;                                                                                                                                    \
+        }                                                                                                                                                           \
+        else                                                                                                                                                        \
+        {                                                                                                                                                           \
+            /* Codes_SRS_TWO_D_ARRAY_07_012: [ TWO_D_ARRAY_FREE_ROW(T) shall free the memory associated with the row specified by row_index and set it to NULL. ]*/ \
+            void* nullptr = NULL;                                                                                                                                   \
+            free((void*)two_d_array->row_arrays[row_index]);                                                                                                        \
+            (void)memcpy((void*)&two_d_array->row_arrays[row_index], &nullptr, sizeof(void*));                                                                      \
```

Comment:
>(void)memcpy((void*)&two_d_array->row_arrays[row_index], &nullptr, sizeof(void*));                                                                      \ [](http://example.com/codeflow?start=12&length=153)

This is an interesting trick. Even for me :-). Why is this needed?

Reply:
removed memcpy